cmd/state-installer/cmd.go

```diff
@@ -192,7 +199,17 @@ func main() {
 func execute(out output.Outputer, cfg *config.Instance, args []string, params *Params) error {
 	// if sourcePath was provided we're already using the right installer, so proceed with installation
 	if params.sourcePath != "" {
-		return installOrUpdateFromLocalSource(out, cfg, args, params)
+		if err := installOrUpdateFromLocalSource(out, cfg, args, params); err != nil {
+			return err
+		}
+		return postInstallEvents(out, params)
+	}
+
+	installPath, _ := installation.InstallPath()
+	// Check if state tool already installed
+	if !params.force && fileutils.TargetExists(appinfo.StateApp(installPath).Exec()) || fileutils.TargetExists(appinfo.StateApp(params.path).Exec()) {
```

Comment:
I think you need parentheses here:

```suggestion
	if !params.force && (fileutils.TargetExists(appinfo.StateApp(installPath).Exec()) || fileutils.TargetExists(appinfo.StateApp(params.path).Exec())) {
```

Reply:
Good eye!